src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx

```diff
@@ -386,7 +386,12 @@ function LineplotViz(props: VisualizationProps<Options>) {
         independentAxisLogScale: false,
         dependentAxisLogScale: false,
         independentAxisValueSpec: 'Full',
-        dependentAxisValueSpec: 'Full',
+        dependentAxisValueSpec:
+          yAxisVar != null
+            ? isSuitableCategoricalVariable(yAxisVar)
```

Comment:
I think you can use `categoricalMode` here (already in the dependencies)

Reply:
@bobular Thank you for your review, test, and suggestion! 👍 The reason why I have used `isSuitableCategoricalVariable` function directly instead of `categoricalMode` is the same reason to your previous implementation/comment, citing "we can't use the more generally scoped 'yAxisVariable' because it's based on vizConfig and is out of date" 😃 It was really helpful to implement my case as well 👍

Reply:
Ahaha, oops. I was clever back then, I guess!

Reply:
Can possibly remove `categoricalMode` from the deps?

Reply:
@bobular I removed the deps and still worked fine. Updated.